static/app/views/performance/traceDetails/newTraceDetailsContent.tsx

```diff
@@ -121,54 +154,123 @@ function NewTraceDetailsContent(props: Props) {
       {roots: 0, orphans: 0}
     );
 
+    if (roots === 0 && orphans > 0) {
+      return TraceType.NO_ROOT;
+    }
+
+    if (roots === 1 && orphans > 0) {
+      return TraceType.BROKEN_SUBTRACES;
+    }
+
+    if (roots > 1) {
+      return TraceType.MULTIPLE_ROOTS;
+    }
+
+    if (orphanErrors && orphanErrors.length > 1) {
+      return TraceType.ONLY_ERRORS;
+    }
+
+    return TraceType.ONE_ROOT;
```

Comment:
nit: should we do `elseif roots === 1` here and throw an error if we get to the else statement? cause for example you could have `roots === 0 and orphans === 0` as a potential case that's currently not captured